experiment/coverage_utils.py

```diff
@@ -45,18 +46,51 @@ def generate_coverage_reports(experiment_config: dict):
     logs.initialize()
     logger.info('Start generating coverage reports.')
 
+    # Create experiment-specific data frames
+    segment_df = pd.DataFrame(
+        columns=["benchmark", "fuzzer", "trial_id", "file_name", "line", "col"])
+    function_df = pd.DataFrame(
+        columns=["benchmark", "fuzzer", "trial_id", "function_name", "hits"])
+    benchmark_name_df = pd.DataFrame(columns=['benchmark_id', 'benchmark'])
+    fuzzer_name_df = pd.DataFrame(columns=['fuzzer_id', 'fuzzer'])
+    filename_df = pd.DataFrame(columns=['file_id', 'file_name'])
+    function_name_df = pd.DataFrame(columns=['function_id', 'function_name'])
+
+    df_container = DataFrameContainer(segment_df, function_df,
+                                      benchmark_name_df, fuzzer_name_df,
+                                      filename_df, function_name_df)
+
     benchmarks = experiment_config['benchmarks']
     fuzzers = experiment_config['fuzzers']
     experiment = experiment_config['experiment']
 
     for benchmark in benchmarks:
         for fuzzer in fuzzers:
-            generate_coverage_report(experiment, benchmark, fuzzer)
+            generate_coverage_report(experiment, benchmark, fuzzer,
+                                     df_container)
 
+    prepare_name_dataframes(df_container)
+    replace_names_with_ids_in_segment_and_function_dataframes(df_container)
+    generate_segment_and_function_csv_files(df_container)
     logger.info('Finished generating coverage reports.')
 
 
-def generate_coverage_report(experiment, benchmark, fuzzer):
+class DataFrameContainer:
+    """Class used for holding DataFrames to extract experiment specific
+    information from  all fuzzer, benchmark and trial combination"""
+
+    # pylint: disable=too-many-arguments
```

Comment:
I think you need to put this on the line where it occurs not on a line by itself.
The way you did it, this check will be disabled for the entire file.